png_layout.py

```diff
@@ -0,0 +1,73 @@
+import pandas as pd
+import layoutparser as lp
+import cv2
+from common_utils import *
+
+# PNG files layout extraction
+class PNG_Layout:
+    def __init__(self, output_folder=os.environ['FOLDER_LAYOUTS']):
+        self.layout_output_folder = output_folder
+
+        self.png_file_list = []
+
+        pl_model_path = "lp://PrimaLayout/mask_rcnn_R_50_FPN_3x/config"
+        pl_labelmap = {1: "TextRegion", 2: "ImageRegion", 3: "TableRegion", 4: "MathsRegion", 5: "SeparatorRegion",
+                       6: "OtherRegion"}
+
+        self.model = lp.Detectron2LayoutModel(config_path=pl_model_path, label_map=pl_labelmap,
+                                            extra_config=["MODEL.ROI_HEADS.SCORE_THRESH_TEST", 0.8])
+
+        if not os.path.exists(self.layout_output_folder):
+            os.makedirs(self.layout_output_folder)
+
+    # PNG layout detection
+    def _png_detect(self, png_file: str):
+        image = cv2.imread(png_file)
+        # print(image)
+        # cv2.imshow("img", image)
+        image = image[..., ::-1]
+
+        layout_pl = self.model.detect(image)
+        # img = lp.draw_box(img, layout_pl, box_width=3,
```

Comment:
Remove all commented-out code. You might keep code that has a comment explaining why it is commented-out and why it is not deleted (e.g., "TODO" experimental code that is "nearly" there but needs some more work or seems to work but needs more testing). Everything else should be removed.

Reply:
fixed